common/laws/00_internal_security.txt

```diff
@@ -1,101 +1,154 @@
 	pop_support = {
 		value = 0
 	}
-	
 	# AI shouldn't try to abolish institutions
 	ai_will_do = {
 		always = no
 	}
-
-	ai_enact_weight_modifier = { #Petitions
+	ai_enact_weight_modifier = {
+		#Petitions
 		value = 0
-		
 		if = {
-			limit = { 
+			limit = {
 				has_journal_entry = je_government_petition
 				has_variable = desired_law_var
-                scope:law = var:desired_law_var
+				scope:law = var:desired_law_var
 			}
 			add = 750
 		}
 	}
 }
 
-law_national_guard = {
+law_provincial_courts = {
 	group = lawgroup_internal_security
-	icon = "gfx/interface/icons/law_icons/national_guard.dds"
-	
-	progressiveness = 25
-
+	icon = "gfx/interface/icons/law_icons/secret_police.dds"
+	progressiveness = 0
 	unlocking_technologies = {
-		law_enforcement
+		centralization
+	}
+	unlocking_laws = {
+		law_monarchy
+		law_autocracy
+		law_oligarchy
+		law_technocracy
+		law_local_police
+		law_dedicated_police
+	}
+	disallowing_laws = {
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
 	}
-
 	institution = institution_home_affairs
 	institution_modifier = {
-		state_conscription_rate_add = 0.005
-		political_movement_radicalism_mult = -0.05
+		country_authority_add = 50
+		country_radicals_from_legitimacy_mult = -0.03
+		country_loyalists_from_legitimacy_mult = 0.03
+		interest_group_ig_landowners_pol_str_mult = 0.10
+		interest_group_ig_petty_bourgeoisie_pol_str_mult = 0.02
+		country_revolution_progress_mult = -0.05
+		country_secession_progress_mult = -0.05
 	}
-	
 	possible_political_movements = {
 		law_secret_police
 		law_guaranteed_liberties
+		law_empowered_gendarmerie
+		law_inquisition
 	}
-
 	pop_support = {
 		value = 0
 		add = {
-			desc = "POP_OFFICERS"
+			desc = "POP_LANDOWNERS"
 			if = {
-				limit = { 
-					is_pop_type = officers
+				limit = {
+					is_pop_type = aristocrats
 				}
 				value = 0.33
 			}
-		}	
+		}
+	}
+	ai_enact_weight_modifier = {
+		#Petitions
+		value = 0
+		if = {
+			limit = {
+				has_journal_entry = je_government_petition
+				has_variable = desired_law_var
+				scope:law = var:desired_law_var
+			}
+			add = 750
+		}
+	}
+}
+
+law_inquisition = {
+	group = lawgroup_internal_security
+	icon = "gfx/interface/icons/law_icons/secret_police.dds"
+	progressiveness = 0
+	unlocking_technologies = {
+		centralization
+	}
+	unlocking_laws = {
+		law_monarchy
+		law_theocracy
+		law_religious_schools
+		law_state_religion
+	}
+	disallowing_laws = {
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
+	}
+	institution = institution_home_affairs
+	institution_modifier = {
+		country_authority_add = 30
+		political_movement_radicalism_mult = -0.10
+		state_conversion_mult = 0.25
+		country_radicals_from_legitimacy_mult = -0.01
+		country_loyalists_from_legitimacy_mult = 0.05
+		interest_group_ig_devout_pol_str_mult = 0.05
+		country_revolution_progress_mult = 0.1
+		country_secession_progress_mult = 0.1
```

Comment:
Either the institution promotes or undermines stability, don't make it confusing by doing both. Quicker progression towards a lower radicalism is confused messaging. Especially since an increased progress speed can have weird side effects, like triggering a revolution anyway.

Example:
83 radicalism, current progress 81, progress step = 18 + 10% = 19.2, it will now cause a revolution on the next update because of the 10%.

Reply:
The Progress Speed is to omit the institution base effect, which grants those. If you load up the game it will show no change as a result.

Since I am merely rescinding a decrease, this effectively does not change the revolution clock at all.